src/track/beatsimporter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <QVector>
+#include <memory>
+
+#include "audio/streaminfo.h"
+
+namespace mixxx {
+
+/// Importer class for Beats that can correct timing offsets when the
+/// signal info (channel number, sample rate, bitrate) is known.
+class BeatsImporter {
+  public:
+    BeatsImporter() = default;
+    virtual ~BeatsImporter() = default;
+
+    virtual bool isEmpty() const {
+        return true;
+    };
+
+    /// Determines the timing offset and returns a Vector of frame positions
+    /// to use as input for the BeatMap constructor
+    virtual QVector<double> importBeatsAndApplyTimingOffset(
```

Comment:
Same here, empty default implementations are dangerous even if they are able to return some result.